From 67a9d693dfcef96be76a1a42e8e6193a86bbc432 Mon Sep 17 00:00:00 2001
From: Pavel Chupin <pavel.v.chupin@intel.com>
Date: Fri, 11 Jan 2013 18:55:01 +0400
Subject: Add -m32/-melf_i386 for x86 target gcc/ld

BZ: 82732

This change simply makes the target arch explicit.
This is necessary for newer versions of gcc, when building on 64-bit hosts,
as the gcc will otherwise assume the target arch is the same as the host.

Bug: ADEV-116
Category: Device Enablement
Domain: AOSP-build
Severity: High
Origin: Internal
Upstream-candidate: Yes

Change-Id: Ifb938b8a2b3c875cf4bcdb04e69c0bc102a40c6b
Signed-off-by: Pavel Chupin <pavel.v.chupin@intel.com>
---
 libc/Android.mk |    7 ++++---
 1 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/libc/Android.mk b/libc/Android.mk
index 208fcc8..7949239 100644
--- a/libc/Android.mk
+++ b/libc/Android.mk
@@ -521,7 +521,8 @@ endif # !arm
 
 ifeq ($(TARGET_ARCH),x86)
   libc_common_cflags += -DSOFTFLOAT
-  libc_crt_target_cflags :=
+  libc_crt_target_cflags := -m32
+  libc_crt_target_ldflags := -melf_i386
   ifeq ($(ARCH_X86_HAVE_SSE2),true)
       libc_crt_target_cflags += -DUSE_SSE2=1
   endif
@@ -651,7 +652,7 @@ ALL_GENERATED_SOURCES += $(GEN)
 GEN := $(TARGET_OUT_INTERMEDIATE_LIBRARIES)/crtbegin_static.o
 $(GEN): $(TARGET_OUT_INTERMEDIATE_LIBRARIES)/crtbegin_static1.o $(TARGET_OUT_INTERMEDIATE_LIBRARIES)/crtbrand.o
 	@mkdir -p $(dir $@)
-	$(hide) $(TARGET_LD) -r -o $@ $^
+	$(hide) $(TARGET_LD) $(libc_crt_target_ldflags) -r -o $@ $^
 ALL_GENERATED_SOURCES += $(GEN)
 
 GEN := $(TARGET_OUT_INTERMEDIATE_LIBRARIES)/crtbegin_dynamic1.o
@@ -666,7 +667,7 @@ ALL_GENERATED_SOURCES += $(GEN)
 GEN := $(TARGET_OUT_INTERMEDIATE_LIBRARIES)/crtbegin_dynamic.o
 $(GEN): $(TARGET_OUT_INTERMEDIATE_LIBRARIES)/crtbegin_dynamic1.o $(TARGET_OUT_INTERMEDIATE_LIBRARIES)/crtbrand.o
 	@mkdir -p $(dir $@)
-	$(hide) $(TARGET_LD) -r -o $@ $^
+	$(hide) $(TARGET_LD) $(libc_crt_target_ldflags) -r -o $@ $^
 ALL_GENERATED_SOURCES += $(GEN)
 
 # We rename crtend.o to crtend_android.o to avoid a
-- 
1.7.4.1

